### PR TITLE
[ctl] Add support for Utilization rules

### PR DESCRIPTION
**Pull request by SokolAndrey**
**What this PR does / why we need it**:
Adds support for Utilization rules for ctl. 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing and/or backwards incompatible change?**:

```release-note
introduces user-facing change 
```

**Does this PR require updating code package or user-facing documentation?**:

```documentation-note

```

### CODE REVIEW

File: src/metrics/rules/validator/validator.go [hunk 1]
```diff
@@ -196,6 +199,54 @@ func (v *validator) validateRollupRules(rrv []view.RollupRule) error {
 	return validateNoDuplicateRollupIDIn(pipelines)
 }
 
+func (v *validator) validateUtilziationRules(rrv []view.UtilizationRule) error {
```
Comment (justinjc) [inline src/metrics/rules/validator/validator.go:202]: Sigh this is so painful. Another perfect case where generics in Go would help here 😕

File: src/metrics/rules/validator/validator.go [hunk 2]
```diff
@@ -196,6 +199,54 @@ func (v *validator) validateRollupRules(rrv []view.RollupRule) error {
...
+				return fmt.Errorf("utilization rule '%s' has invalid pipeline '%v': %v", rule.Name, target.Pipeline, err)
+			}
+
+			// Validate that the storage policies are valid.
+			if err := v.validateStoragePolicies(target.StoragePolicies, types); err != nil {
+				return fmt.Errorf("utilization rule '%s' has invalid storage policies in %v: %v", rule.Name, target.StoragePolicies, err)
+			}
+			pipelines = append(pipelines, target.Pipeline)
+		}
+		//TODO 22/01/2021 andreisokol: any other validations?
```
Comment (justinjc) [inline src/metrics/rules/validator/validator.go:244]: The only real difference I see for utilization rules compared to roll up rules is the addition of `Value`. Not sure what this is used for to say whether a validation on it makes sense though.

File: src/metrics/rules/utilization.go [hunk 3]
```diff
@@ -0,0 +1,406 @@
...
+)
+
+var (
+	errNoRollupTargetsInUtilizationRuleSnapshot = errors.New("no rollup targets in utilization rule snapshot")
+	errUtilizationRuleSnapshotIndexOutOfRange   = errors.New("utilization rule snapshot index out of range")
+	errNilUtilizationRuleSnapshotProto          = errors.New("nil utilization rule snapshot proto")
+	errNilUtilizationRuleProto                  = errors.New("nil utilization rule proto")
+)
+
+// rollupRuleSnapshot defines a rule snapshot such that if a metric matches the
```
Comment (justinjc) [inline src/metrics/rules/utilization.go:42]: Update comment copypasta

File: go.mod [hunk 4]
```diff
@@ -36,7 +36,8 @@ require (
 	github.com/golang/mock v1.4.4
 	github.com/golang/protobuf v1.4.2
 	github.com/golang/snappy v0.0.1
-	github.com/google/go-cmp v0.5.2
+	github.com/golangci/golangci-lint v1.35.2 // indirect
+	github.com/google/go-cmp v0.5.4
```
[the remark this answers is not in the document]
Comment (SokolAndrey) [reply go.mod:40]: no, probably was updated by one of the `make gen` commands

File: src/metrics/rules/view/utilization.go [hunk 5]
```diff
@@ -0,0 +1,44 @@
+package view
+
+type UtilizationRule struct {
+	ID                  string        `json:"id,omitempty`
+	Name                string        `json:"name" validate:"required"`
+	Tombstoned          bool          `json:"tombstoned"`
+	CutoverMillis       int64         `json:"cutoverMillis,omitempty"`
+	Value               string        `json:"value,omitempty"`
+	Filter              string        `json:"filter" validate:"required"`
+	Targets             rollupTargets `json:"targets" validate:"required,dive,required"`
```
[the remark this answers is not in the document]
Comment (SokolAndrey) [reply src/metrics/rules/view/utilization.go:10]: I used the same structure as `RollupRule`. It's defined in [rollup](https://github.com/m3db/m3/blob/b70fd4478d343a78b7464999af8bfc00b06a4eb1/src/metrics/rules/view/rollup.go#L77:6) I don't really see why we need this type at all. I'll change it to `[]RollupTarget`

**Comment by SokolAndrey** [comment]
Duplicate of #3154

**Pull request closed without merging**